.github/workflows/ci-all-in-one-build.yml

```diff
@@ -21,58 +19,78 @@ jobs:
-      run: |
-        case ${GITHUB_EVENT_NAME} in
-          pull_request)
-            echo "PR_ONLY=pr-only" >> ${GITHUB_ENV}
-            ;;
-          *)
-            echo "PR_ONLY=" >> ${GITHUB_ENV}
-            ;;
-        esac
-
-    - name: Build, test, and publish all-in-one image
-      run: |
-        BINARY=${{ matrix.mode.binary }} \
-          SKIP_SAMPLING=${{ matrix.mode.skip_sampling }} \
-          bash scripts/build-all-in-one-image.sh ${{ env.PR_ONLY }}
-      env:
-        DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
-        QUAY_TOKEN: ${{ secrets.QUAY_TOKEN }}
+      - name: Harden Runner
+        uses: step-security/harden-runner@63c24ba6bd7ba022e95695ff85de572c04a18142 # v2.7.0
+        with:
+          egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          submodules: true
+
+      - name: Fetch git tags
+        run: |
+          git fetch --prune --unshallow --tags
+
+      - uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: 1.22.x
+
+      - name: Setup Node.js version
+        uses: ./.github/actions/setup-node.js
+
+      - name: Export BRANCH variable
+        uses: ./.github/actions/setup-branch
+
+      - name: Install tools
+        run: make install-ci
+
+      - uses: docker/setup-qemu-action@68827325e0b33c7199eb31dd4e31fbe9023e06e3 # v3.0.0
+
+      - name: Define PR_ONLY var if running on a Pull Request
+        run: |
+          case ${GITHUB_EVENT_NAME} in
+            pull_request)
+              echo "PR_ONLY=pr-only" >> ${GITHUB_ENV}
+              ;;
+            *)
+              echo "PR_ONLY=" >> ${GITHUB_ENV}
+              ;;
+          esac
+
+      - name: Build, test, and publish all-in-one image
+        run: |
+          BINARY=${{ matrix.mode.binary }} \
+            SKIP_SAMPLING=${{ matrix.mode.skip_sampling }} \
+            bash scripts/build-all-in-one-image.sh ${{ env.PR_ONLY }}
+        env:
+          DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+          QUAY_TOKEN: ${{ secrets.QUAY_TOKEN }}
+
+      - name: Set up Docker Compose
+        run: sudo curl -L "https://github.com/docker/compose/releases/download/v2.11.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose && sudo chmod +x /usr/local/bin/docker-compose
+
+      - name: Start Docker Compose services
+        run: docker-compose -f docker-compose/monitor/docker-compose.yml up -d
+
+      - name: Run smoke tests
+        run: |
+          # Wait for services to be up
+          sleep 30
+          # Check if Jaeger UI is accessible
+          curl -f http://localhost:16686 || exit 1
+          # Check if Prometheus is accessible
+          curl -f http://localhost:9090 || exit 1
+          # Check if Grafana is accessible
+          curl -f http://localhost:3000 || exit 1
```

Comment:
The objective of e2e test is to verify that Jaeger components are fulfilling the expected function. In case of SPM, we can query the internal API that returns the time series results to the UI and validate that there are data points in those results.